pilot/pkg/features/pilot.go

```diff
@@ -566,6 +566,11 @@ var (
 		false,
 		"If enabled, controllers required for ambient will run. This is required to run ambient mesh.").Get()
 
+	EnableNodeUntaintControllers = env.Register(
+		"PILOT_ENABLE_NODE_UNTAINT_CONTROLLERS",
+		false,
+		"If enabled, controller that untaints nodes with cni pods ready will run. This should be enabled if you disabled ambient init containers.").Get()
+
```

Comment:
If there is controversial on former API on this, we could enable it as an env var like you proposed here first.